File: backend/app/schemas/curriculum.py
from pydantic import BaseModel
from typing import Optional
from datetime import date, time

class SubjectBase(BaseModel):
    name: str
    description: Optional[str] = None
    teacher_id: int

class SubjectCreate(SubjectBase):
    pass

class SubjectUpdate(SubjectBase):
    name: Optional[str] = None
    teacher_id: Optional[int] = None

class SubjectResponse(SubjectBase):
    id: int
    teacher_name: str  # 添加教师姓名字段

    class Config:
        orm_mode = True

class BookBase(BaseModel):
    name: str
    description: Optional[str] = None
    subject_id: int
    order: int

class BookResponse(BookBase):
    id: int

    class Config:
        orm_mode = True

class ChapterBase(BaseModel):
    name: str
    description: Optional[str] = None
    book_id: int
    order: int

class ChapterResponse(ChapterBase):
    id: int

    class Config:
        orm_mode = True

class SectionBase(BaseModel):
    name: str
    description: Optional[str] = None
    chapter_id: int
    order: int

class SectionResponse(SectionBase):
    id: int

    class Config:
        orm_mode = True

class ClassBase(BaseModel):
    subject_id: int
    teacher_id: int
    name: str
    date: date
    start_time: time
    end_time: time
    description: Optional[str] = None

class ClassResponse(ClassBase):
    id: int
    teacher_name: str  # 添加教师姓名字段

    class Config:
        orm_mode = True 

File: frontend/src/types/curriculum.ts
export interface Subject {
  id: number;
  name: string;
  description: string;
  teacher_id: number;
  teacher_name: string;
}

export interface Book {
  id: number;
  subject_id: number;
  name: string;
  description: string;
  order: number;
}

export interface Chapter {
  id: number;
  book_id: number;
  name: string;
  description: string;
  order: number;
}

export interface Section {
  id: number;
  chapter_id: number;
  name: string;
  description: string;
  order: number;
}

export interface Class {
  id: number;
  subject_id: number;
  teacher_id: number;
  teacher_name: string;
  name: string;
  date: string;
  start_time: string;
  end_time: string;
  description: string;
} 